cmd/common/image/build.go

```diff
@@ -17,80 +17,203 @@ package image
 import (
 	"bytes"
 	"context"
+	_ "embed"
 	"encoding/json"
 	"errors"
 	"fmt"
 	"io"
 	"os"
+	"os/exec"
+	"path/filepath"
 
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/oci"
 	"github.com/opencontainers/image-spec/specs-go"
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
 	"oras.land/oras-go/v2"
 	"oras.land/oras-go/v2/content"
 	"oras.land/oras-go/v2/errdef"
+)
 
-	"github.com/inspektor-gadget/inspektor-gadget/pkg/oci"
+//go:embed build.sh
+var buildScript []byte
+
+const (
+	DEFAULT_BUILDER_IMAGE = "ghcr.io/inspektor-gadget/inspektor-gadget-ebpf-builder:latest"
 )
 
-type buildOptions struct {
-	progAmd64FilePath  string
-	progArm64FilePath  string
-	definitionFilePath string
-	image              string
+type buildFile struct {
+	EBPFSource string `yaml:"ebpfsource"`
+	Definition string `yaml:"definition"`
+	CFlags     string `yaml:"cflags"`
+	// TODO: custom build script
+	// TODO: author, etc.
```

Comment:
It does make sense for some gadgets, but it should be rare: some syscalls only exist in one architecture, so some gadgets might only make sense in one architecture.

In my opinion, this could be added in another PR.

Reply:
I agree to handle it in another PR, I don't think we need that right now.